### PR TITLE
Consider realm of card when determining if realm is editable

### PR DESCRIPTION
**Pull request by habdelra**
This PR replaces the `SessionService` with a `RealmResource`. The issue with the `SessionService` is that it incorrectly assumed that only a single realm session was necessary, when in fact we need a realm session for any realm that we access. This resulted in obtaining an incorrect realm session for any card in operator mode that did not originate from the "default realm" of the host app (the realm that the host app is being served from). The default realm is not special, it's one of many realms that will require a session from the host app.

TODO:
- [x] add new tests for stacks from differing realms with differing permissions

### CODE REVIEW

File: packages/host/tests/acceptance/interact-submode-test.gts [hunk 1]
```diff
@@ -50,17 +54,21 @@ module('Acceptance | interact submode tests', function (hooks) {
...
+  setupMatrixServiceMock(hooks, () => realmPermissions);
 
   hooks.afterEach(async function () {
-    window.localStorage.removeItem('recent-cards');
-    window.localStorage.removeItem('recent-files');
+    windowMock.localStorage.removeItem('recent-cards');
+    windowMock.localStorage.removeItem('recent-files');
+    // intentionally not using the window mock because otherwise the actual
+    // app's local storage stomps on top of the window mock
+    window.localStorage.removeItem('boxel-session');
```
Comment (backspace) [inline packages/host/tests/acceptance/interact-submode-test.gts:62]: The way the library is designed to be used is to import `window` where it’s going to be used, then it’s the real one in development/production and the mock on in tests.

**Review by backspace — CHANGES_REQUESTED** [review]
Thanks for figuring this out, I think it would make sense to standardise on using the window mock everywhere vs having a mixed approach, though maybe there‘s something I’m not understanding about this context.

File: packages/host/tests/acceptance/interact-submode-test.gts (hunk 1 above)
Comment (habdelra) [reply packages/host/tests/acceptance/interact-submode-test.gts:62]: so i think the problem is that in the `resources/realm.ts` I'm not importing the mock. I'll add that import there

**Comment by habdelra** [comment]
@backspace your issue was addressed in  https://github.com/cardstack/boxel/pull/1026/commits/642efeb5228b578315a614a7361284a2f4006dd0, please check again

**Review by backspace — APPROVED** [review]
great, thanks!

**Review by FadhlanR — APPROVED** [review]

File: packages/host/app/resources/realm.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+import type MatrixService from '@cardstack/host/services/matrix-service';
+
+interface Args {
+  named: {
+    realmURL: URL;
+  };
+}
+
+type RealmJWT = TokenClaims & { iat: number; exp: number };
```
Comment (jurgenwerk) [inline packages/host/app/resources/realm.ts]: We already have this type, we can import it from here:

https://github.com/cardstack/boxel/blob/da8392578752aafbedd34427ee017b6ea6dd0426/packages/runtime-common/realm-auth-client.ts#L6

**Review by jurgenwerk — APPROVED** [review]

**Pull request merged**